src/quick-lint-js/fe/parse-statement.cpp

```diff
@@ -4825,15 +4826,8 @@ void Parser::parse_and_visit_import(
   // import myns = ns;           // TypeScript only.
   // import C = ns.C;            // TypeScript only.
   case Token_Type::equal:
-    if (possibly_typescript_import_alias) {
-      if (!this->options_.typescript) {
-        this->diag_reporter_->report(
-            Diag_TypeScript_Import_Alias_Not_Allowed_In_JavaScript{
-                .import_keyword = import_span,
-                .equal = this->peek().span(),
-            });
-      }
-
+    if (possibly_typescript_import_namespace_alias) {
+      std::optional<Source_Code_Span> equal_span = this->peek().span();
```

Comment:
Nit: Why is this optional? I think just `Source_Code_Span` would work well:
```suggestion
      Source_Code_Span equal_span = this->peek().span();
```

Reply:
Done!